external/storm-hdfs/src/main/java/org/apache/storm/hdfs/avro/DefinedAvroSerializer.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.hdfs.avro;
+
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaNormalization;
+import org.apache.commons.codec.binary.Base64;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DefinedAvroSerializer extends AbstractAvroSerializer {
+
+    private final static String FP_ALGO = "CRC-64-AVRO";
+    final Map<String, Schema> fingerprint2schemaMap = new HashMap<>();
+    final Map<Schema, String> schema2fingerprintMap = new HashMap<>();
+
+    public DefinedAvroSerializer() throws IOException, NoSuchAlgorithmException {
```

Comment:
@abhishekagarwal87 In this case, getting the pre-defined schemas from a file felt like the right thing to do.  A user could specify an arbitrary number of schemas, each of arbitrary size.  That didn't feel right to put in the global storm config.  Happy to hear other opinions on that though.  Thanks!